cli/tsc/97_ts_host.js

```diff
@@ -789,33 +763,98 @@ export function filterMapDiagnostic(diagnostic) {
       }
     }
   }
+
   return true;
 }
 
 // list of globals that should be kept in Node's globalThis
-ts.deno.setNodeOnlyGlobalNames([
-  "__dirname",
-  "__filename",
-  "Buffer",
-  "BufferConstructor",
-  "BufferEncoding",
-  "clearImmediate",
-  "clearInterval",
-  "clearTimeout",
-  "console",
-  "Console",
-  "ErrorConstructor",
-  "gc",
-  "Global",
-  "localStorage",
-  "queueMicrotask",
-  "RequestInit",
-  "ResponseInit",
-  "sessionStorage",
-  "setImmediate",
-  "setInterval",
-  "setTimeout",
+ts.deno.setNodeOnlyGlobalNames(
+  new Set([
+    "__dirname",
+    "__filename",
+    '"buffer"',
+    "Buffer",
+    "BufferConstructor",
+    "BufferEncoding",
+    "clearImmediate",
+    "clearInterval",
+    "clearTimeout",
+    "console",
+    "Console",
+    "crypto",
+    "ErrorConstructor",
+    "gc",
+    "Global",
+    "localStorage",
+    "queueMicrotask",
+    "RequestInit",
+    "ResponseInit",
+    "sessionStorage",
+    "setImmediate",
+    "setInterval",
+    "setTimeout",
+  ]),
+);
+// List of globals in @types/node that collide with Deno's types.
+// When the `@types/node` package attempts to assign to these types
+// if the type is already in the global symbol table, then assignment
+// will be a no-op, but if the global type does not exist then the package can
+// create the global.
+const setTypesNodeIgnorableNames = new Set([
+  "AbortController",
+  "AbortSignal",
+  "AsyncIteratorObject",
+  "atob",
+  "Blob",
```

Comment:
How did you generate that list?

Reply:
I added logging for whenever the `@types/node` project assigned to a global then put everything in here then deleted the ones I didn't think were necessary. Probably some of these could still be removed. It actually might be better for us to use an allow list, but I'm not sure.

Reply:
I'm more worried about getting this list again in X months.

Reply:
Let's see how this goes. I mostly just kept it to anything that was causing errors with our globals so maybe we can remove or add to this over time based on feedback and errors we encounter when upgrading.